generator/resources/eks_daemon_test_matrix.json

```diff
@@ -0,0 +1,5 @@
+[
```

Comment:
So that the GenerateMatrix job can generate _something_, albeit right now this just uses the default values for the terraform vars. This might change, but I guess it depends how we structure different tests for different launch types

Reply:
Do you have an example of what the GenerateMatrix job would generate?

Reply:
https://github.com/SaxyPandaBear/amazon-cloudwatch-agent/actions/runs/4926249422/jobs/8801541681#step:5:23 So this takes a combination of the `generator/resources/ec2_linux_test_matrix.json`, and basically just does a dot product (yay math) with the list of test directory paths configured. In `generator/test_case_generator.go`, I see:
```go
	"ec2_linux": {
		"./test/ca_bundle",
		"./test/cloudwatchlogs",
		"./test/metrics_number_dimension",
		"./test/metric_value_benchmark",
		"./test/run_as_user",
		"./test/collection_interval",
		"./test/metric_dimension",
	},
```

So for each matrix configuration, which defines OS types, etc, each of them will be paired with each test directory above. This results in this gigantic JSON blob:
```json
[ { test_dir: ./test/ca_bundle, os: ubuntu-20.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/cloudwatchlogs, os: ubuntu-20.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metrics_number_dimension, os: ubuntu-20.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metric_value_benchmark, os: ubuntu-20.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/run_as_user, os: ubuntu-20.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/collection_interval, os: ubuntu-20.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metric_dimension, os: ubuntu-20.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/ca_bundle, os: ubuntu-22.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu-LTS-22*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/cloudwatchlogs, os: ubuntu-22.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu-LTS-22*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metrics_number_dimension, os: ubuntu-22.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu-LTS-22*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metric_value_benchmark, os: ubuntu-22.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu-LTS-22*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/run_as_user, os: ubuntu-22.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu-LTS-22*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/collection_interval, os: ubuntu-22.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu-LTS-22*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metric_dimension, os: ubuntu-22.04, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ubuntu-LTS-22*, binaryName: amazon-cloudwatch-agent.deb, username: ubuntu, installAgentCommand: go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/ca_bundle, os: debian-10, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-10-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/cloudwatchlogs, os: debian-10, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-10-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metrics_number_dimension, os: debian-10, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-10-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metric_value_benchmark, os: debian-10, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-10-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/run_as_user, os: debian-10, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-10-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/collection_interval, os: debian-10, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-10-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metric_dimension, os: debian-10, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-10-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/ca_bundle, os: debian-11, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-11-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/cloudwatchlogs, os: debian-11, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-11-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metrics_number_dimension, os: debian-11, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-11-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metric_value_benchmark, os: debian-11, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-11-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/run_as_user, os: debian-11, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-11-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/collection_interval, os: debian-11, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-11-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/metric_dimension, os: debian-11, family: linux, testType: ec2_linux, arc: arm64, instanceType: c6g.large, ami: cloudwatch-agent-integration-test-debian-11-arm64*, binaryName: amazon-cloudwatch-agent.deb, username: admin, installAgentCommand: /snap/bin/go run ./install/install_agent.go deb, caCertPath: /etc/ssl/certs/ca-certificates.crt, values_per_minute: 0 }, { test_dir: ./test/ca_bundle, os: al2, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-al2*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/cloudwatchlogs, os: al2, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-al2*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/metrics_number_dimension, os: al2, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-al2*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/metric_value_benchmark, os: al2, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-al2*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/run_as_user, os: al2, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-al2*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/collection_interval, os: al2, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-al2*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/metric_dimension, os: al2, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-al2*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/ca_bundle, os: rhel8, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-rhel8-base*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/cloudwatchlogs, os: rhel8, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-rhel8-base*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/metrics_number_dimension, os: rhel8, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-rhel8-base*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/metric_value_benchmark, os: rhel8, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-rhel8-base*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/run_as_user, os: rhel8, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-rhel8-base*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/collection_interval, os: rhel8, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-rhel8-base*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/metric_dimension, os: rhel8, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-rhel8-base*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/ca_bundle, os: ol9, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ol9*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/cloudwatchlogs, os: ol9, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ol9*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/metrics_number_dimension, os: ol9, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ol9*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/metric_value_benchmark, os: ol9, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ol9*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/run_as_user, os: ol9, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ol9*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/collection_interval, os: ol9, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ol9*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 }, { test_dir: ./test/metric_dimension, os: ol9, family: linux, testType: ec2_linux, arc: amd64, instanceType: t3a.medium, ami: cloudwatch-agent-integration-test-ol9*, binaryName: amazon-cloudwatch-agent.rpm, username: ec2-user, installAgentCommand: go run ./install/install_agent.go rpm, caCertPath: /etc/ssl/certs/ca-bundle.crt, values_per_minute: 0 } ]
```